src/raftstore/store/snap.rs

```diff
@@ -61,10 +115,82 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileReader: Read {
+    fn exists(&self) -> bool;
+    fn path(&self) -> String;
+    fn meta(&self) -> io::Result<Metadata>;
+    fn total_size(&self) -> io::Result<u64>;
+    fn delete(&self);
+}
+
+pub trait RecvSnapshotFileReader {
+    fn path(&self) -> String;
```

Comment:
The only usage of path is for displaying the path in log message. If it returns `&Path`, we need to call `path().display()` everywhere it's called. It would be simple to just cast it to the string here.
Another reason for the return value to be string is that for the new format, there would be 3 cf files displayed like `gen_1_1_1_(default|write|lock).tmp` for short. It's not existent path and generated on the fly.